sympy/polys/numberfields.py

```diff
@@ -517,6 +517,8 @@ def _minpoly_compose(ex, x, dom):
     if ex is I:
         _, factors = factor_list(x**2 + 1, x, domain=dom)
         return x**2 + 1 if len(factors) == 1 else x - I
+    if ex is GoldenRatio:
+        return x**2 - x - 1
```

Comment:
I think we need the same factorization check as above with `I` to see if phi is already included in the domain.